### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by J-C-S-V**
Hello dear reviewer 👋, I hope you're having a nice day today 😊

This is what I did on this branch:

- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend.
- Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make my HTTP requests to the backend
- Fetch my books from the Bookstore API after the component has rendered
- Refactor my add book and remove book features to persist your changes in the server

Best regards! ✨

### CODE REVIEW

File: src/redux/features/books/booksSlice.js [hunk 1]
```diff
@@ -1,48 +1,72 @@
...
+  // console.log('delete', id);
+  return id;
+});
+
+// {
+//   item_id: 'item1',
+//   title: 'The Great Gatsby',
+//   author: 'John Smith',
+//   category: 'Fiction',
+// }
```
Comment (omar-muhamad) [inline src/redux/features/books/booksSlice.js]: - [x] Kindly delete all commented(unused) code in your app files to make your code more readable and clean, also It's a good practice to keep your app free of any unused code.

File: src/redux/features/books/booksSlice.js [hunk 2]
```diff
@@ -1,48 +1,72 @@
...
+  return books;
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  try {
+    await axios.post(`${url}${urlId}/books`, book);
+  } catch (error) {
+    throw new Error(error);
+  }
+  // console.log('post', book);
```
Comment (omar-muhamad) [inline src/redux/features/books/booksSlice.js]: - [x] Nice job using try and catch with API calls, but please make your code clean of any commented(unused) code. (don't forget to remove console logs after you have done with them)

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- App functionality works as expected 👌.
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by J-C-S-V** [comment]
Hello @Omar-Muhamad, 👋

Thank you for the feedback, I'll keep an eye on that on my next projects.

Regards!!!

**Review by cynthiainga — APPROVED** [review]
Hi @J-C-S-V, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [x] Correct Gitflow ✔️ 
- [x] No linters errors ✔️ 
- [x] The app work as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by J-C-S-V** [comment]
Thank you!!! @cynthiainga

**Pull request merged**